### PR TITLE
Web Server + Factory Dao, Factory Server 

### PR DESCRIPTION
**Pull request by AleksandraPonochovna**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (AleksandraPonochovna) [inline pom.xml]: It's a full project. HW 19 + HW 20 + HW 21
Next Homework i will do so like you said (i will use some other branches)

Comment (AleksandraPonochovna) [reply pom.xml]: Sorry for the inconvenience

Comment (AleksandraPonochovna) [reply pom.xml]: It's my mistake, but now i can't change it

File: src/main/java/collector/AddProductServlet.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/add/product")
+public class AddProductServlet extends HttpServlet {
```
Comment (SevostyanovGerman) [inline src/main/java/collector/AddProductServlet.java]: cs

File: src/main/java/collector/AddProductServlet.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    private ProductService productService = ProductServiceFactory.getProductService();
+    private Product product;
+    private Long id = 1L;
```
Comment (SevostyanovGerman) [inline src/main/java/collector/AddProductServlet.java]: move to

File: src/main/java/collector/AddProductServlet.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    @Override
+    protected void doPost(HttpServletRequest request,
+                          HttpServletResponse response) throws IOException, ServletException {
+        try {
+            String name = request.getParameter("name");
+            String description = request.getParameter("description");
+            Double price = Double.valueOf(request.getParameter("price"));
+            product = new Product(id, name, description, price);
+            productService.addProduct(product);
+            id++;
```
Comment (SevostyanovGerman) [inline src/main/java/collector/AddProductServlet.java]: ??

File: src/main/java/collector/AddUserServlet.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/add/user")
+public class AddUserServlet extends HttpServlet {
+
+    private UserService userService = UserServiceFactory.getUserService();
+    private User user;
+    private Long id = 1L;
```
Comment (SevostyanovGerman) [inline src/main/java/collector/AddUserServlet.java]: -=-

File: src/main/java/collector/AddUserServlet.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    protected void doPost(HttpServletRequest request,
+                          HttpServletResponse response) throws IOException, ServletException {
+        try {
+            String email = request.getParameter("email");
+            String password = request.getParameter("password");
+            String repeatPassword = request.getParameter("rpassword");
+            if (password.equals(repeatPassword)) {
+                user = new User(id, email, password);
+                userService.addUser(user);
+                id++;
```
Comment (SevostyanovGerman) [inline src/main/java/collector/AddUserServlet.java]: -=-

File: src/main/java/collector/AllProductsServlet.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/products")
+public class AllProductsServlet extends HttpServlet {
+
+    private static final ProductService productService = ProductServiceFactory.getProductService();
+    @Override
```
Comment (SevostyanovGerman) [inline src/main/java/collector/AllProductsServlet.java]: cs

File: src/main/java/collector/AllUsersServlet.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@WebServlet(value = "/users")
+public class AllUsersServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getUserService();
+    private List<User> allUsers = userService.getAllUsers();
+    @Override
```
Comment (SevostyanovGerman) [inline src/main/java/collector/AllUsersServlet.java]: cs

File: src/main/java/collector/AllUsersServlet.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@WebServlet(value = "/users")
+public class AllUsersServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getUserService();
+    private List<User> allUsers = userService.getAllUsers();
```
Comment (SevostyanovGerman) [inline src/main/java/collector/AllUsersServlet.java]: ?

File: src/main/java/collector/SignInServlet.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@WebServlet(value = "/sign")
+public class SignInServlet extends HttpServlet {
+
+    private UserService userService = UserServiceFactory.getUserService();
+    private static final String TEST_EMAIL = "admin@ru";
```
Comment (SevostyanovGerman) [inline src/main/java/collector/SignInServlet.java]: create init servlet which create test user

File: src/main/java/collector/SignInServlet.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+        String password = request.getParameter("password");
+        List<User> allUsers = userService.getAllUsers();
+        if (!allUsers.isEmpty()) {
+            for (User user : allUsers) {
+                if (email.equals(user.getEmail()) && password.equals(user.getPassword())) {
+                    request.getRequestDispatcher("/users.jsp").forward(request, response);
+                }
+            }
+        } else {
+            if (email.equals(TEST_EMAIL) && password.equals(TEST_PASSWORD)){
```
Comment (SevostyanovGerman) [inline src/main/java/collector/SignInServlet.java]: del

File: src/main/java/dao/impl/UsersDaoImpl.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @Override
+    public List<User> getAllUsers() {
+        return Database.users;
+    }
+
+    @Override
+    public void deleteUser(Long id) {
+        for (User user : Database.users) {
```
Comment (SevostyanovGerman) [inline src/main/java/dao/impl/UsersDaoImpl.java]: stream?

File: src/main/java/collector/AddProductServlet.java (hunk 2 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/AddProductServlet.java]: ok

File: src/main/java/collector/AddProductServlet.java (hunk 3 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/AddProductServlet.java]: I created class IdGeneratorUtil in package util

File: src/main/java/collector/AddProductServlet.java (hunk 4 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/AddProductServlet.java]: I deleted it

File: src/main/java/collector/AddUserServlet.java (hunk 5 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/AddUserServlet.java]: i deleted

File: src/main/java/collector/AddUserServlet.java (hunk 6 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/AddUserServlet.java]: i deleted

File: src/main/java/collector/AllProductsServlet.java (hunk 7 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/AllProductsServlet.java]: ok

File: src/main/java/collector/AllUsersServlet.java (hunk 8 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/AllUsersServlet.java]: ok

File: src/main/java/collector/AllUsersServlet.java (hunk 9 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/AllUsersServlet.java]: i deleted

File: src/main/java/collector/SignInServlet.java (hunk 11 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/SignInServlet.java]: Ok

File: src/main/java/collector/SignInServlet.java (hunk 10 above)
Comment (AleksandraPonochovna) [reply src/main/java/collector/SignInServlet.java]: look! i added method init()
is it ok?

**Pull request merged**